### PR TITLE
chore: add Snyk scans to docs

### PR DESCRIPTION
**Pull request by crenshaw-dev**
We currently maintain Snyk scans in the Snyk web UI.

This has two downsides:
1) It's difficult to keep the web UI in sync with the current repo state (latest versions, latest ignore rules, latest image tags from manifests).
2) It's not public.

By generating the scans weekly and placing them in the docs, we can give users a reasonably up-to-date picture of the state of the currently-maintained versions of Argo CD.

The new page looks like this:

image

Each link goes to a scan detail page that looks like this:

image

### CODE REVIEW

File: hack/snyk-report.sh [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+Every Sunday, Snyk scans are generated for Argo CD's \`master\` branch and the most recent patches of the three most
+recent minor releases.
+
+The [latest](https://argo-cd.readthedocs.io/en/latest/snyk/) version of this page shows the most recent scan.
+The [stable](https://argo-cd.readthedocs.io/en/stable/snyk/) version of this page shows the last scan before the most
+recent stable release was cut. The latest version will have the most up-to-date information.
+
+## Scans
+EOM
+
```
Comment (jannfis) [inline hack/snyk-report.sh:15]: Want to do all that follows in a temporary directory created by `mktemp` maybe and then move files to the location of the real repository?

**Review by jannfis — COMMENTED** [review]
Wow! This is awesome, @crenshaw-dev. Great work.

I have a few comments, which are not formal requests for change, just thoughts that came to my mind.

File: .github/workflows/update-snyk.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Build reports
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+        run: |
+          make snyk-report
+          git config --global user.email 'ci@argoproj.com'
+          git config --global user.name 'CI'
+          git commit -am "Update Snyk reports"
```
[the remark this answers is not in the document]
Comment (crenshaw-dev) [reply .github/workflows/update-snyk.yaml:19]: Good thinking. I'll restrict the `add` glob as carefully as is practical.

And agreed, the latest stable branch would be a better destination!

**Comment by crenshaw-dev** [comment]
@jannfis can you take another look?

File: .github/workflows/update-snyk.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build reports
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+        run: |
+          stable_release="release-$(sed -E 's/\.[0-9]+$//g' VERSION)"
+          git checkout "$stable_release"
+          make snyk-report
```
Comment (jannfis) [inline .github/workflows/update-snyk.yaml:20]: Shall the cronjob run off the stable release branch or off `master`?

I think it should _run_ off the `master` branch (so we don't have to cherry-pick changes to the script), and store the results in the stable branch. :thinking:

Comment (crenshaw-dev) [reply .github/workflows/update-snyk.yaml:20]: Ah wow, good point. Will fix.

**Comment by crenshaw-dev** [comment]
@jannfis the `git checkout` for the target branch now happens in the script instead of in the workflow. Can you take another look?

File: mkdocs.yml [hunk 4]
```diff
@@ -2,9 +2,9 @@ extra_css:
 - assets/versions.css
 extra_javascript:
 - assets/versions.js
-google_analytics:
-- UA-105170809-2
-- auto
+#google_analytics:
+#- UA-105170809-2
+#- auto
```
Comment (jannfis) [inline mkdocs.yml:7]: Also a leftover from testing?

**Review by jannfis — APPROVED** [review]
LGTM with some minor nits and/or questions

Thanks @crenshaw-dev !

File: .github/workflows/update-snyk.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build reports
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+        run: |
+          stable_release="release-$(sed -E 's/\.[0-9]+$//g' VERSION)"
```
[the remark this answers is not in the document]
Comment (jannfis) [reply .github/workflows/update-snyk.yaml]: Oops. Old review comment. Please ignore :)

File: .github/workflows/update-snyk.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+          make snyk-report target_branch="$stable_release"
+          git config --global user.email 'ci@argoproj.com'
+          git config --global user.name 'CI'
+          git add docs/snyk/index.md
+          git add docs/snyk/*/*.html
+          git commit -m "[Bot] Update Snyk reports"
+          git push --set-upstream crenshaw-dev
```
[the remark this answers is not in the document]
Comment (crenshaw-dev) [reply .github/workflows/update-snyk.yaml]: Good catch. Fixed!

**Comment by jetersen** [comment]
Some of these path names lead to:

```
Cloning into 'argo-cd'...
remote: Enumerating objects: 90785, done.
remote: Counting objects: 100% (465/465), done.
remote: Compressing objects: 100% (272/272), done.
remote: Total 90785 (delta 259), reused 341 (delta 185), pack-reused 90320
Receiving objects: 100% (90785/90785), 68.47 MiB | 16.73 MiB/s, done.
Resolving deltas: 100% (56895/56895), done.
error: invalid path 'docs/snyk/master/ghcr.io_dexidp_dex:v2.32.0.html'
fatal: unable to checkout working tree
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry with 'git restore --source=HEAD :/'

exit status 128
```

**Pull request merged**